src/Common/src/Interop/Windows/secur32/ApplicationProtocols.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace System.Net
+{
+    // _SEC_APPLICATION_PROTOCOLS
+    [StructLayout(LayoutKind.Sequential,Pack = 1)]
+    internal struct ApplicationProtocols
+    {
+        private static readonly int ProtocolListOffset = Marshal.SizeOf<ApplicationProtocols>();
+        private static readonly int ProtocolListConstSize = ProtocolListOffset - (int)Marshal.OffsetOf<ApplicationProtocols>("ProtocolExtenstionType");
+        public uint ProtocolListsSize;
+        public ApplicationProtocolNegotiationExt ProtocolExtenstionType;
+        public short ProtocolListSize;
+
+        public static unsafe byte[] ToByteArray(string[] applicationProtocols)
+        {
+            long protocolListSize = 0;
+            foreach (string applicationProtocol in applicationProtocols)
+            {
+                if (string.IsNullOrEmpty(applicationProtocol)) throw new ArgumentException(SR.net_ssl_app_protocols_invalid, "applicationProtocols");
```

Comment:
Who creates the applicationProtocols array?  Is it coming from external to our implementation, or is this an array we supply and validate such that we'd never expect to see such errors in production?  If it's just internal, we can make these Debug.Asserts.  If it's not, we should split this check apart to throw an ArgumentNullException in the case of null rather than throwing the base ArgumentException for that.